h2o-genmodel/src/main/java/hex/genmodel/utils/ParseUtils.java

```diff
@@ -39,10 +49,16 @@ public static Object tryParse(String input, Object defVal) {
 
         try { return Integer.parseInt(input); }
         catch (NumberFormatException e) {
-            if ((defVal instanceof Number) && ! (defVal instanceof Double || defVal instanceof Float))
+            if ((defVal instanceof Number) && !(defVal instanceof Double || defVal instanceof Float || defVal instanceof Long))
                 throw e; // integer number expected but couldn't be parsed
         }
 
+        try { return Long.parseLong(input); }
+        catch (NumberFormatException e) {
+            if (defVal instanceof Number)
```

Comment:
should this be `if ((defVal instanceof Number) && ! (defVal instanceof Double || defVal instanceof Float))`?